File: TSDW/Esercitazioni/itinere3/game_l/app/Http/Controllers/PlayerController.php
<?php

namespace App\Http\Controllers;

use App\Models\Player;
use Illuminate\Http\Request;

class PlayerController extends Controller
{
    /**
     * Display a listing of the resource.
     */
    public function index()
    {
        return view("players.index", ["players"=>Player::all()]);
    }

    /**
     * Show the form for creating a new resource.
     */
    public function create()
    {
        return view("players.create");
    }

    /**
     * Store a newly created resource in storage.
     */
    public function store(Request $request)
    {
        Player::create($request->all());
        return redirect()->route('players.index');
    }
    /**
     * Display the specified resource.
     */
    public function show(Player $player)
    {
        return view('players.show', ['player'=>$player]);
    }
    /**
     * Show the form for editing the specified resource.
     */
    public function edit(Player $player)
    {
        return view('players.edit', ['player'=>$player]);
    }

    /**
     * Update the specified resource in storage.
     */
    public function update(Request $request, Player $player)
    {
        $player->update($request->all());
        return redirect()->route('players.index');
    }

    /**
     * Remove the specified resource from storage.
     */
    public function destroy(Player $player)
    {
        $player->delete();
        return redirect()->route('players.index');
    }
}

File: TSDW/Esercitazioni/itinere3/game_s/src/main/java/com/example/game_s/controllers/PlayerController.java
package com.example.game_s.controllers;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;
import org.springframework.ui.Model;
import com.example.game_s.models.Player;
import com.example.game_s.repositories.PlayerRepository;

@Controller
public class PlayerController {
    
    private final PlayerRepository repo;

    public PlayerController(PlayerRepository repo){
        this.repo = repo;
    }
    
    @GetMapping("/")
    public String home(){
        return "index";
    }

    @GetMapping("/players/create")
    public String create(){
        return "players/create";
    }

    @GetMapping("/players")
    public String index(Model model){
        model.addAttribute("players", repo.findAll());  // Cambiato "player" a "players" per evitare confusione
        return "players/index";  // Cambiato "players/create" a "players/index"
    }

    @PostMapping("/players/store")
    public String store(Player player){
        repo.save(player);
        return "redirect:/players";  // Cambiato "redirect:players" a "redirect:/players"
    }

    @GetMapping("/players/edit")
    public String show(Model model, Long id){  // Cambiato l'annotazione da @PostMapping a @GetMapping
        model.addAttribute("player", repo.findById(id).orElse(null));
        return "players/edit";
    }

    @PostMapping("/players/update")
    public String update(Player player){
        repo.save(player);
        return "redirect:/players";  // Cambiato "redirect:players" a "redirect:/players"
    }

    @PostMapping("/players/delete")
    public String delete(Long id){
        repo.deleteById(id);
        return "redirect:/players";  // Cambiato "redirect:players" a "redirect:/players"
    }
}
